tests/test_magic.rs

```diff
@@ -0,0 +1,105 @@
+use deku::prelude::*;
+use hexlit::hex;
+use rstest::rstest;
+use std::convert::{TryFrom, TryInto};
+
+pub mod samples {
+    use super::*;
+
+    #[derive(PartialEq, Debug, DekuRead, DekuWrite)]
+    #[deku(magic = b"deku")]
+    pub struct MagicDeku {}
+
+    #[derive(PartialEq, Debug, DekuRead, DekuWrite)]
+    #[deku(magic = b"deku", type = "u8")]
+    pub enum EnumMagicDeku {
+        #[deku(id = "0")]
+        Variant,
+    }
+
+    #[derive(PartialEq, Debug, DekuRead, DekuWrite)]
+    #[deku(magic = b"UKED")]
+    pub struct NestedMagicDeku {
+        pub nested: MagicDeku,
+    }
+}
+
+#[rstest(input,
+    case(&hex!("64656b75")),
+
+    #[should_panic(expected = "Parse(\"Missing magic value [100, 101, 107, 117]\")")]
```

Comment:
Are the multiple "Missing magic value" tests testing different code paths?

Reply:
They are each testing a different byte of the input being incorrect, which I figured was worth checking for to future-proof against really bad implementations of magic reading that, say, only check for the last byte and assume the first few are correct.

(Not saying that specific case is going to happen, obviously, but I just figured it was better to be safe than sorry)